### PR TITLE
Don't allow to add listening topics without audios

### PR DESCRIPTION
**Pull request by Anlord241**

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/services/impl/RestrictionsServiceImpl.java [hunk 1]
```diff
@@ -202,4 +203,11 @@ public void checkNotArchivedContentFile(ContentFile contentFile) {
             throw new QuestionOrTopicEditingException("You can't edit archived topics");
         }
     }
+
+    @Override
+    public void checkListeningHasAudio(ContentFile contentFile) {
+        if (contentFile.getUrl() == null) {
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/RestrictionsServiceImpl.java:214]: Let's also check that the file contains on s3

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

File: src/main/java/com/team4/testingsystem/services/impl/RestrictionsServiceImpl.java [hunk 2]
```diff
@@ -202,4 +207,19 @@ public void checkNotArchivedContentFile(ContentFile contentFile) {
...
+            throw new NoAudioException("You can't add listening topics without audios");
+        }
+    }
+
+    @Override
+    public void checkExistsOnS3(String fileName) {
+        Path filePath = Path.of(FileUtils.getTempDirectory() + "/testing-system/" + fileName);
```
Comment (maxim12321) [inline src/main/java/com/team4/testingsystem/services/impl/RestrictionsServiceImpl.java]: I don't think this check is correct. Temp files aren't stored permanently, that's why they are temp :)

**Review by maxim12321 — CHANGES_REQUESTED** [review]

File: src/main/java/com/team4/testingsystem/services/impl/RestrictionsServiceImpl.java (hunk 2 above)
Comment (maxim12321) [reply src/main/java/com/team4/testingsystem/services/impl/RestrictionsServiceImpl.java]: Amazon S3 SDK has this method:
`boolean doesObjectExist(String bucketName, String objectName)`
I think that's what you need

Comment (AntonMaksimenkau) [reply src/main/java/com/team4/testingsystem/services/impl/RestrictionsServiceImpl.java]: We have several filestorages (at least local one and s3). And they have different logic for such checks. I think we should add isFileExist method to the interface and implement it on all the file storages. Then we should call the method here and throw the exception if file is not exist

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

File: src/main/java/com/team4/testingsystem/services/FilesService.java [hunk 3]
```diff
@@ -11,4 +11,6 @@
     Resource load(String fileName) throws FileLoadingFailedException;
 
     void delete(String fileName) throws FileDeletingFailedException;
+
+    boolean isFileExist(String fileName);
```
Comment (maxim12321) [inline src/main/java/com/team4/testingsystem/services/FilesService.java]: Maybe `doesFileExist`?)

**Review by maxim12321 — APPROVED** [review]

**Pull request merged**